### PR TITLE
Change the JavaScript package manager from `yarn` back to `npm`

### PR DESCRIPTION
**Pull request by Gargron**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -47,20 +47,19 @@ RUN apt-get update && \
...
-COPY Gemfile* package.json yarn.lock /opt/mastodon/
+COPY Gemfile* package.json package-lock.json /opt/mastodon/
 
 RUN cd /opt/mastodon && \
   bundle config set --local deployment 'true' && \
   bundle config set --local without 'development test' && \
   bundle config set silence_root_warning true && \
 	bundle install -j"$(nproc)" && \
-	yarn install --pure-lockfile
+	npm install
```
Comment (ykzts) [inline Dockerfile:62]: `npm ci` behaves the same as `yarn install --pure-lockfile`.

https://docs.npmjs.com/cli/v8/commands/npm-ci

Comment (shleeable) [reply Dockerfile:62]:
```
rails assets:precompile
```
requires yarn?

File: Dockerfile [hunk 2]
```diff
@@ -113,7 +112,7 @@ USER mastodon
 # Precompile assets
 RUN cd ~ && \
 	OTP_SECRET=precompile_placeholder SECRET_KEY_BASE=precompile_placeholder rails assets:precompile && \
```
Comment (shleeable) [inline Dockerfile:114]:
```
rails assets:precompile
```
requires yarn?

**Comment by Gargron** [comment]
Looks like webpacker requires yarn, never mind

**Pull request closed without merging**